runtime/onert/api/nnfw/src/nnfw_api_internal.cc

```diff
@@ -1356,6 +1365,17 @@ NNFW_STATUS nnfw_session::train_set_traininfo(const nnfw_train_info *info)
     _train_info->setBatchSize(info->batch_size);
     _train_info->setLossInfo(loss_info);
     _train_info->setOptimizerInfo(opt_info);
+
+    for (uint32_t idx = 0; idx < info->trainble_ops_size; ++idx)
+    {
+      if (idx >= primary_subgraph()->operations().size())
+      {
+        std::cerr << "Error during nnfw_session::train_set_traininfo: provided op_index=" << idx
+                  << " is out of operators range" << std::endl;
+        return NNFW_STATUS_ERROR;
+      }
+      _train_info->enableTrainNodeUpdate(onert::ir::OperationIndex{idx});
+    }
```

Comment:
Is there a way to remove values from `_trainable_ops`?

Reply:
I see applied, removed also `enableTrainNodeUpdate`, `disableTrainNodeUpdate` to avoid keeping the state in other places